core/system/src/mpu/vmpu_freescale_k64_mem.c

```diff
@@ -60,17 +126,13 @@ void vmpu_mem_switch(uint8_t src_box, uint8_t dst_box)
 
     box = &g_mem_box[dst_box];
 
-    /* for box zero, just disable private ACL's */
-    src_count = g_mem_box[src_box].count;
-    if(src_box && !dst_box)
-    {
-        /* disable private regions */
-        if(src_count)
-        {
-            t = g_mem_acl_user + 1;
-            while(src_count--)
-                MPU->WORD[t++][3] = 0;
-        }
+    /* Disable all user MPU regions. */
+    for (t = K64F_MPU_REGIONS_USER; t < K64F_MPU_REGIONS_MAX; t++) {
+        MPU->WORD[t][3] = 0;
```

Comment:
@niklas-arm instead of the array-assignment - in case there's not a struct for a single MPU slot provided by Freescale, please create one and use names instead of array indexes (with volatile members etc.)